PHPCompatibility/Tests/Syntax/NewDynamicClassConstantFetchUnitTest.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * PHPCompatibility, an external standard for PHP_CodeSniffer.
+ *
+ * @package   PHPCompatibility
+ * @copyright 2012-2020 PHPCompatibility Contributors
+ * @license   https://opensource.org/licenses/LGPL-3.0 LGPL3
+ * @link      https://github.com/PHPCompatibility/PHPCompatibility
+ */
+
+namespace PHPCompatibility\Tests\Syntax;
+
+use PHPCompatibility\Tests\BaseSniffTestCase;
+
+/**
+ * Test the NewDynamicClassConstantFetch sniff.
+ *
+ * @group newDynamicClassConstantFetch
+ * @group syntax
+ *
+ * @covers \PHPCompatibility\Sniffs\Syntax\NewDynamicClassConstantFetchSniff
+ *
+ * @since 10.0.0
+ */
+final class NewDynamicClassConstantFetchUnitTest extends BaseSniffTestCase
+{
+
+    /**
+     * Ensure a warning message when found syntax on earlier versions.
+     *
+     * @dataProvider dataDynamicClassConstantFetch
+     *
+     * @param int $line The line number.
+     *
+     * @return void
+     */
+    public function testDynamicClassConstantFetch($line)
+    {
+        $file = $this->sniffFile(__FILE__, '8.2');
+        $this->assertError($file, $line, 'Dynamic class constant fetch is not available in PHP 8.2 or earlier.');
+    }
+
+    /**
+     * Data provider.
+     *
+     * @return array
+     * @see    testDynamicClassConstantFetch()
+     */
+    public static function dataDynamicClassConstantFetch()
+    {
+        return [
+            [17],
+            [18],
+            [19],
+            [20],
+            [23],
+        ];
+    }
+
+    /**
+     * Ensure a warning message in NOT found syntax on supported versions.
+     *
+     * @dataProvider dataNoViolationsOnValidVersion
+     *
+     * @param int $line The line number.
+     *
+     * @return void
+     */
+    public function testNoViolationsOnValidVersion($line)
+    {
+        $file = $this->sniffFile(__FILE__, '8.3');
+        $this->assertNoViolation($file, $line);
+    }
+
+    /**
+     * Data provider.
+     *
+     * @return array
+     * @see    testNoViolationsOnValidVersion()
+     */
+    public static function dataNoViolationsOnValidVersion()
+    {
+        return [
+            [17],
+            [18],
+            [19],
+            [20],
+            [23],
+        ];
+    }
+
+    /**
+     * Ensure a warning message in NOT found in syntax that is not explicitly targeted by this sniff.
+     *
+     * @dataProvider dataNoFalsePositives
+     *
+     * @param int $line The line number.
+     *
+     * @return void
+     */
+    public function testNoFalsePositives($line)
+    {
+        $file = $this->sniffFile(__FILE__, '8.3');
```

Comment:
For this test, we want to make sure that with a `testVersion` set to a PHP version on which the syntax was **_not_** yet supported, the sniff does not throw false positives for code it shouldn't flag.

That means the `testVersion` needs to be set to `8.2` (or lower), as with `8.3` the sniff will (should) not flag anything at all, so wouldn't actually be tested against "no false positives".

```suggestion
        $file = $this->sniffFile(__FILE__, '8.2');
```